lib/tdf3/src/tdf.ts

```diff
@@ -982,77 +991,44 @@ async function unwrapKey({
     };
   }
 
-  // Get unique split IDs to determine if we have an OR or AND condition
-  const splitIds = new Set(Object.keys(splitPotentials));
-
-  // If we have only one split ID, it's an OR condition
-  if (splitIds.size === 1) {
-    const [splitId] = splitIds;
+  const poolSize = concurrencyLimit === undefined ? 1 : concurrencyLimit > 1 ? concurrencyLimit : 1;
```

Comment:
Can we break this down so it's more readable? Nested ternary's are too hard to parse, in my opinion.

Reply:
better?